homework/App.js

```diff
@@ -17,13 +17,42 @@ class App {
     // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
 
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
+    const header = Util.createAndAppend('header', root, { class: 'header' });
+    Util.createAndAppend('p', header, { text: 'HYF Repositories', class: 'hyf' });
+    const selectMenu = Util.createAndAppend('select', header, { class: 'repository-selector' });
+    const containerDiv = Util.createAndAppend('div', root, { id: 'container' });
+    const repoDiv = Util.createAndAppend('div', containerDiv, { class: 'left-div' });
+    const table = Util.createAndAppend('table', repoDiv);
+
+    const contributorsDiv = Util.createAndAppend('div', containerDiv, { class: 'right-div' });
+    Util.createAndAppend('p', contributorsDiv, {
+      text: 'Contributors',
+      class: 'contributor-header',
+    });
+    const contributorList = Util.createAndAppend('ul', contributorsDiv, {
+      class: 'contributor-list',
+      id: 'contributorListId',
+    });
 
     try {
       const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      this.repos = repos
+        .sort((a, b) => a.name.localeCompare(b.name, 'en', { sensivity: 'base' }))
+        .map(repo => new Repository(repo));
```

Comment:
I see two of `new Repository` a one is here an another in line [46], makes one of them redundant, But in the same time if you removed one of them you will get your applications halted.

Let's break it down:
 - the constructor of `Repository` class takes a `repository` _singular_
   - > means this line is correct some how, so we will see on the way down.
 - with the singular value that class takes there is no need for the `index` on the render method in the class in `./Repository.js`

Looking to what do we have here as a `new Repository` initialized and the previous steps makes lines 46 and 47 an extra step and constructing another complete Object of other content, because you have to pass in a singular value, while you are passing an array of repositories making them plural.

So if we commented them out of the way then we will have a bit more freedom of not changing the original code of the application **(Important after the problem solved for more details)**

Now how to apply our changes without breaking things?


**please follow the next comment...**

Reply:
Thank you for very helpful and detailed feedback. I changed the code as you advised.